src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/client/util/HomeOfficeApiUtilTest.java

```diff
@@ -0,0 +1,136 @@
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.AsylumCaseDefinition;
+
+@SpringJUnitConfig
+@ExtendWith(MockitoExtension.class)
+public class HomeOfficeApiUtilTest {
+
+    private HomeOfficeApiUtil homeOfficeApiUtil;
+
+    private ObjectMapper objectMapper;
+
+    @Value("classpath:home-office-sample-response.json")
+    private Resource resource;
+
+    @BeforeEach
+    void setUp() {
+        homeOfficeApiUtil = new HomeOfficeApiUtil();
+    }
+
+    @Test
+    public void create_home_office_header_returns_valid() {
+        HttpHeaders headers = homeOfficeApiUtil.getHomeOfficeHeader();
+
+        assertNotNull(headers);
+        assertThat(headers.getContentType()).isEqualTo(MediaType.APPLICATION_JSON);
+    }
+
+    @Test
+    public void createMessageHeader_returns_valid_values() {
+        Map<String, Object> messageHeader = homeOfficeApiUtil.createMessageHeader();
+
+        assertNotNull(messageHeader);
+        assertThat(messageHeader.size()).isEqualTo(3);
+        assertNotNull(messageHeader.get("consumer"));
+        assertNotNull(messageHeader.get("correlationId"));
+        assertNotNull(messageHeader.get("eventDateTime"));
+    }
+
+    @Test
+    public void createConsumer_returns_valid_values() {
+        Map<String, String> consumer = homeOfficeApiUtil.createConsumer();
+
+        assertNotNull(consumer);
+        assertThat(consumer.size()).isEqualTo(2);
+        assertThat(consumer.get("code")).isEqualTo("HMCTS");
+        assertThat(consumer.get("description")).isEqualTo("HM Courts and Tribunal Service");
+    }
+
+    @Test
+    public void call_extract_home_office_data_should_return_all_fields() {
+
+        try (Reader reader = new InputStreamReader(resource.getInputStream(), UTF_8)) {
+            String jsonResponse = FileCopyUtils.copyToString(reader);
+
+            AsylumCase asylumCase = homeOfficeApiUtil.extractHomeOfficeData(jsonResponse);
+
+            assertNotNull(asylumCase);
+            assertEquals(Optional.of("Smith"),
+                asylumCase.read(AsylumCaseDefinition.HO_APPELLANT_FAMILY_NAME));
+            assertEquals(Optional.of("Capability"),
+                asylumCase.read(AsylumCaseDefinition.HO_APPELLANT_GIVEN_NAME));
+            assertEquals(Optional.of("CAN (denoting Canada)"),
+                asylumCase.read(AsylumCaseDefinition.HO_APPELLANT_NATIONALITY_CODE));
+            assertEquals(Optional.of("Canada"),
+                asylumCase.read(AsylumCaseDefinition.HO_APPELLANT_NATIONALITY));
+        } catch (JsonProcessingException e) {
+            throw new UncheckedIOException(e);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+    }
+
+    @Test
+    public void call_extract_home_office_data_should_return_null_for_error() {
+
+        try {
+            AsylumCase asylumCase = homeOfficeApiUtil.extractHomeOfficeData(getErrorJsonResponseString());
+            assertThat(asylumCase).isEmpty();
```

Comment:
Are you expecting an empty asylum case here?  Should that happen?

Reply:
Modified to throw exception if we do not get good response